plugins/destination.go

```diff
@@ -156,6 +156,9 @@ func (p *DestinationPlugin) Write(ctx context.Context, tables schema.Tables, sou
 			Data:      schema.TransformWithTransformer(p.client, r.Data),
 		}
 		select {
+		case <-gctx.Done():
```

Comment:
I think that was the culprit. the line below is not needed as I used `ctx` instead of `gctx`. If the main context will be cancelled then `gctx` also will be cancelled.

Reply:
Ah yeah, good point